src/txpasslib/_threads.py

```diff
@@ -0,0 +1,24 @@
+from twisted.internet.defer import Deferred
+from twisted.python.failure import Failure
+
+
+def defer_to_worker(deliver, worker, work, *args, **kwargs):
+    """
+    Run a task in a worker, delivering the result as a ``Deferred`` in the
+    reactor thread.
+    """
+    deferred = Deferred()
+
+    def wrapped_work():
+        try:
+            result = work(*args, **kwargs)
+        except BaseException:
+            f = Failure()
+            deliver(lambda: deferred.errback(f))
```

Comment:
`deliver` will be a `callFromThread` method from an [`IReactorFromThreads`](https://twistedmatrix.com/documents/16.6.0/api/twisted.internet.interfaces.IReactorFromThreads.html) provider, so its signature will be `callable, *args, **kwargs`.  That means you could eliminate the `lambda` and write this instead:

```python
deliver(deferred.errback, f)
```